apps/web/components/eventtype/EventAdvancedTab.tsx

```diff
@@ -401,7 +401,14 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupInfered
                     selectedCustomInput.type = customInput.type;
                     selectedCustomInput.options = customInput.options || undefined;
                     // Update by id
-                    const inputIndex = customInputs.findIndex((input) => input.id === values.id);
+                    const inputIndex = customInputs.findIndex(
+                      (input) =>
+                        input.id === values.id &&
```

Comment:
This seems hackish. id should uniquely identify an element, why do we need to match all props

Reply:
so id is only assigned after user has clicked save button otherwise a default id is being used -1. so editing multiple custom inputs without saving would work something like this https://photos.google.com/share/AF1QipMCbabiovDIRlru0ASfB0uerBr-aoN4XLG0_xTwAerAzAm1FFeRA4Kp-yf-Yi4bWA/photo/AF1QipP42jgzOHOeraQ45gF0MFcHCRxcEhjfl1dfyMq1?key=ZkZ0ZnZhQ2d0UXNYdE8wSHFmaWJjNTdsaWdGRmp3

It would overwrite other inputs which have same id.

Reply:
Can we generate id(from uuid()) at frontend ? That should solve the problem.

Reply:
so i tried generating random id ( can't use uuid() because we want id of type number. i tried something else here).  but a lot of logic in https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/eventTypes.tsx#L38 depends on id being <0 ( -1 which is default).

Reply:
@hariombalhara  okay i was facing some bugs earlier now i am using Math.random to generate id because type required is number only.

added a new optional variable which is being used to detect if custom input has to be created or updated .Earlier we were using id<0 (-1 which is default) to check new custom inputs https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/eventTypes.tsx#L38 

and lastly useEffect to update customInput after saving.

this should work now